base/float.jl

```diff
@@ -14,7 +14,7 @@ const NaN = NaN64
 ## conversions to floating-point ##
 convert(::Type{Float16}, x::Integer) = convert(Float16, convert(Float32,x))
 for t in (Int8,Int16,Int32,Int64,Int128,UInt8,UInt16,UInt32,UInt64,UInt128)
-    @eval promote_rule(::Type{Float16}, ::Type{$t}) = Float32
+    @eval promote_rule(::Type{Float16}, ::Type{$t}) = Float16
```

Comment:
Yes I should have noted that this is https://github.com/JuliaLang/julia/pull/17297